docstrings fix, try #2

jvanasco opened the pull request:
This is a second version of #579

JonathanHuot commented on oauthlib/oauth2/rfc6749/grant_types/base.py:
```diff
@@ -116,14 +116,32 @@ def register_code_modifier(self, modifier):
+        """
+        :param request: OAuthlib request.
+        :type request: oauthlib.common.Request
+        :param token_handler: A token handler instace, for example of type
```
Typo error `instace` i/o `instance` in all token_handler locations :-)

jvanasco replied:
ack! sorry! dealing with the merge-commit messages was so messy, so I just redid all the work.

JonathanHuot commented on oauthlib/oauth2/rfc6749/parameters.py:
```diff
@@ -133,15 +132,19 @@ def prepare_token_revocation_request(url, token, token_type_hint="access_token",
-    automatically.  This specification defines two such values:
+    :param token: REQUIRED.  The token that the client wants to get revoked.
+
+    param:token_type_hint: OPTIONAL.  A hint about the type of the token
```
`param:`  instead of `:param `

jvanasco commented:
I fixed those two. I also updated the formatting on the `errors` file, and updated anything that uses the format "see methodname" to be "see ``.methodname``", which (I believe) should render better on automatic doc generators.

Since there wasn't an official styleguide, I adhered to the following existing conventions as I standardized stuff.  None of these are of my own design, they already existed but weren't applied universally. I just tried to make things consistent.

1. params in the docs are listed in their order as args/kwargs
2. if there is a textual docstring, use a single blank line between the docstring and the params.
3. I put referenced class functions in backticks as I found them. (likely many more)
4. Every `:param foo:` line is considered a sentence and ends in a period. Every `:type bar:` line is considered an object reference and does not end in a period.

Finally,  if i didn't know what the param's doc was, I still included it but left it empty -- for someone to fill out later.  You can just regex for ":param (\w+):\n" to find them.

JonathanHuot commented:
Could you add these instructions in the contribution guide documentation part itself ?

I think it does not harm to have this clearly stated in our documentation. That's very good to list the changes you made and that would be a shame to lose it.

jvanasco commented:
absolutely. i'll push an update asap.

jvanasco commented:
This PR closes #578
